server/middleware/auth.js

```diff
@@ -0,0 +1,43 @@
+const {verify} = require('../helper/jwt')
+const {User, Todo} = require('../models')
+
+const authentication = (req,res,next) =>{
+    const {access_token} = req.headers
+        const user = verify(access_token)
+        req.user = user
+        console.log(user,"ini nihh")
+        User.findOne({where:{email:user.email}})
+        .then(data=>{
+            if (data){
+                next()
+            }else{
+                throw ({message:`user not authenticate`,status:401})
+            }
+        })
+        .catch(err=>{
+            next(err)
+        })
+}
+
+const authorization = (req,res,next)=>{
+    const id = req.params.id
+    Todo.findOne({where:{id}})
+    .then(data=>{
+        if (data){
+            if(data.UserId === req.user.id){
+                next()
+            }else{
+                throw ({message:`forbiden access`,status:403})
+            }
+        }else{
+            throw ({message:`forbiden access`,status:403})    
```

Comment:
yang ini `404` saja..  not found, agar ketika debug, bisa tahu dengan lebih detail kemungkinan errornya pa